### PR TITLE
Fix issue with rasa.data.is_story_file where NLU training files were passing check

### LINKED ISSUE #5992: Internal bug: NLU files pass the rasa.data.is_story_data check
**Rasa version**: 1.10.2 (master)

**Python version**: 3.7.5

**Operating system** (windows, osx, ...): Linux

**Issue**:
Markdown files (even in nlu format) still pass the is_story_file test (because of the loose regex in `rasa.data._contains_story_pattern`).

**Solution**:
- [x] Test `rasa.data.is_story_file` with NLU training data formats, assert they are not story files
- [x]  Make regex more specific and don't match on NLU training data formats

### PR DESCRIPTION
**Pull request by ortsaCniveK**
**Proposed changes**:
- Updates the `_contains_story_pattern` regex to be more specific and checks for NLU training data formats.

Fixes #5992.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @tmbo will take a look at it as soon as possible ✨

File: changelog/5992.bugfix.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix `rasa.data.is_story_file` check whereby Markdown files in NLU training data format were passing this check. Now only story files will pass this check.
```
Comment (tmbo) [inline changelog/5992.bugfix.md:1]: this file needs to be `.rst`, do you mind changing the extension?

Comment (tmbo) [inline changelog/5992.bugfix.md:1]:
```suggestion
Fix ``rasa.data.is_story_file`` check whereby Markdown files in NLU training data format were passing this check. Now only story files will pass this check.
```

**Comment by tmbo** [comment]
I'll leave the actual code review to @degiz now that we merged the changes to the story training data format

**Comment by ortsaCniveK** [comment]
Ah, well if that's the case, then there might be no need for this internal bug fix then, but I'll leave it to you @degiz !

**Comment by tmbo** [comment]
the detection changed now that we use yaml formats, I think it is safer to keep the old format the way it is and rather push people to use yaml instead.

**Pull request closed without merging**